torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp

```diff
@@ -255,12 +259,11 @@ void RegisterInplaceNodeInLoopBlocks(Value* orig_data, Value* new_data) {
     node_list.pop_back();
   }
 
-  // Update inplace node inputs inside the inner most block.
-  auto prev_data = inplace_node->inputs().at(0);
-  while (IsInplaceNode(prev_data->node())) {
-    prev_data = prev_data->node()->inputs().at(0);
-  }
-  for (auto node : inplace_node->owningBlock()->nodes()) {
+  // Update inplace node inputs inside the outer most block.
+  auto outer_block_node = outer_block->owningNode();
+  auto prev_data =
```

Comment:
why loop for prev_data no longer needed?

Reply:
At this point in the pass, the node to be updated in the block is the last input added to the block.
So we don't need to iterate over inplace op inputs to find it. We can directly access the last block input.